src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@SubJunk I'm not aware of any convention for that, I just try to find a way to make the code readable. I'm not very happy with the way it looks either, but all the other things I've tried have looked worse in my opinion.

Some quick googling shows that some others use this form as well, but there doesn't seem to be much consensus. The Eclipse formatter uses this formatting:
```java
sleepMode = getBoolean(KEY_OLD_PREVENTS_SLEEP, true) 
	? PreventSleepMode.PLAYBACK
	: PreventSleepMode.NEVER;
```
Do you think that is better/more readable? While I find it pretty clean, I think putting the ```?``` on the next line can make it easy to miss that it *is* a ternary in the first place.

I like to use descriptive identifiers - the downside is that lines tend to get long, so I've had to split ternaries in several places and would prefer if we had an "agreed upon" way to format them.